### PR TITLE
Update client for db

### PR DESCRIPTION
**Pull request by samau3**
1. Update how client handles JWT due to change in what is stored
2. Destructure JWT in server so it doesn't have as many property levels to access
3. Update return types for database queries to better represent what is being returned
4. Update typing for some parameters and functions
5. Need to explore different error handling options for Profile page

### CODE REVIEW

File: app/src/pages/Profile/index.tsx [hunk 1]
```diff
@@ -18,8 +18,32 @@ import UserContext from "../../UserContext";
...
+
+    // make requests to get username and balance, transactions
+    // add a try catch in case accessing db has issues
+    useEffect(() => {
+        async function loadPortfolio() {
+            try {
+                const userProfile = await PaperTraderApi.getUserProfile();
+                setPortfolio(userProfile);
+            } catch (error) {
+                logout(); // need to change - considering react error boundary
```
Comment (samau3) [inline app/src/pages/Profile/index.tsx:42]: Tried to do something similar to how errors are handled for Login component, but didn't behave how I wanted to. Looking to try using React Error boundaries, but not putting that in this PR so it doesn't get too big.

File: app/src/pages/Login/index.tsx [hunk 2]
```diff
@@ -6,11 +6,10 @@ import UserContext from "../../UserContext";
 
 import UserSession from '../../helpers/UserSession';
 import { DISCORD_REDIRECT_URI } from "../../config";
-import { AxiosError } from 'axios';
 import { CsrfStateError } from '../../errors/errors';
 
-interface handleLogin {
-  handleLogin: () => Promise<void>;
+interface login {
```
Comment (li-andy) [inline app/src/pages/Login/index.tsx:11]: types are capitalized by convention

Comment (li-andy) [reply app/src/pages/Login/index.tsx:11]: `LoginParams` would work

File: app/src/pages/Profile/index.tsx (hunk 1 above)
Comment (li-andy) [reply app/src/pages/Profile/index.tsx:42]: > but didn't behave how I wanted to.

can you elaborate on this?

File: server/src/routes/profile.ts [hunk 3]
```diff
@@ -12,6 +13,9 @@ export const router = Router();
 router.get('/', async (req, res, next) => {
   try {
     const userPortfolio = await User.getProfile(res.locals.user);
+    if (userPortfolio === undefined) {
+      throw new BadRequestError("No user found")
```
Comment (li-andy) [inline server/src/routes/profile.ts:17]: seems more like a `NotFoundError`?

File: app/src/pages/Login/index.tsx [hunk 4]
```diff
@@ -28,7 +27,7 @@ interface handleLogin {
  * App --> Login
  */
 
-function Login({ handleLogin }: handleLogin) {
+function Login({ login }: login) {
```
[the remark this answers is not in the document]
Comment (samau3) [reply app/src/pages/Login/index.tsx:30]: Ah thought handleLogin was a weird name, but having a Login component and login function is more confusing; will change it back

File: app/src/App.tsx [hunk 5]
```diff
@@ -29,25 +26,27 @@ interface UserData {
...
+	const [currentUser, setCurrentUser] = useState<string | undefined>(undefined);
 	const [errors, setErrors] = useState<string | undefined>(undefined)
 	const [searchParams] = useSearchParams();
 
 	/** 
-	  * Stores string generated from PaperTraderApi in localStorage to
-	  * use for Discord OAUth CSRF prevention when component mounts and updates 
-	  * currentUser state if there userToken changes from default/previous value
-	  */
+		* Stores string generated from PaperTraderApi in localStorage to
```
[the remark this answers is not in the document]
Comment (samau3) [reply app/src/App.tsx:35]: Will have to resolve in the next PR that has mass reformatting

File: app/src/pages/Login/index.tsx (hunk 2 above)
Comment (samau3) [reply app/src/pages/Login/index.tsx:11]: > types are capitalized by convention

Types or interfaces?  

Also in this specific case, should it be `type` or `login` when assigning the type to the component prop?

Comment (li-andy) [reply app/src/pages/Login/index.tsx:11]: 1. both
1. interface

File: server/src/routes/profile.ts (hunk 3 above)
Comment (samau3) [reply server/src/routes/profile.ts:17]: Good point! Made the change.

File: app/src/pages/Profile/index.tsx (hunk 1 above)
Comment (samau3) [reply app/src/pages/Profile/index.tsx]: There's three scenarios that I'm aware of that will cause an error during `User.getProfile()`:

1. Database is down
2. Server is down
3. Database data changed between user sessions (e.g. user was removed from the database due to DROP DATABASE)

And the corresponding error handling:

1. Stay on profile page, update error state, display error message
2. Call logout and redirect to /login page, update error state, display error message
3. Call logout and redirect to /login page, update error state, display error message

With the way I handled errors previously using `try/catch` it works for the first situation but not the other two as the `Profile` component is unmounted upon calling `logout` and I can't change the state once the component is unmounted. The component unmounting and being unable to change the error state is the behavior I wasn't expecting, but is expected behavior for React.

So the solutions I wanted to explore are:

- React Error Boundary (unlikely to be the complete solution as this is not a component rendering issue)
- Upon redirect to `/login` I send a message from the `Profile` that will be used to update the `Login` component's error state to display

Attempting these solutions in a separate PR as this current PR was starting to grow.

**Review by li-andy — APPROVED** [review]

File: app/src/pages/Login/index.tsx (hunk 2 above)
Comment (samau3) [reply app/src/pages/Login/index.tsx]: Determined `interface` is still more appropriate as my use case doesn't require `type`

**Pull request merged**